File: Python/encapsulation.py
class Animal:
    def __init__(self, name, age):
        self.__name = name  # Private attribute
        self.__age = age    # Private attribute

    # Getter for name
    def get_name(self):
        return self.__name

    # Setter for name
    def set_name(self, name):
        self.__name = name

    # Getter for age
    def get_age(self):
        return self.__age

    # Setter for age
    def set_age(self, age):
        self.__age = age

# Create an instance of Animal
dog = Animal("Buddy", 3)
print(f"Name: {dog.get_name()}, Age: {dog.get_age()}")
dog.set_name("Max")
dog.set_age(4)
print(f"Updated Name: {dog.get_name()}, Updated Age: {dog.get_age()}")

File: examples and misc/Cpp/encapsulation.cpp
#include <iostream>
using namespace std;

class Animal {
private:
    string name;
    int age;

public:
    // Constructor
    Animal(string n, int a) : name(n), age(a) {}

    // Getter for name
    string getName() {
        return name;
    }

    // Setter for name
    void setName(string n) {
        name = n;
    }

    // Getter for age
    int getAge() {
        return age;
    }

    // Setter for age
    void setAge(int a) {
        age = a;
    }
};

int main() {
    Animal dog("Buddy", 3);
    cout << "Name: " << dog.getName() << ", Age: " << dog.getAge() << endl;
    dog.setName("Max");
    dog.setAge(4);
    cout << "Updated Name: " << dog.getName() << ", Updated Age: " << dog.getAge() << endl;
    return 0;
}
